src/js/parsers/captions/parsers.dfxp.js

```diff
@@ -56,9 +56,10 @@ define([
             utils.tryCatch(function() {
                 var idx = data.indexOf('begin=\"');
                 data = data.substr(idx + 7);
```

Comment:
Overwriting "data" is what causes it to not work when end is before begin. To make it work in all cases, we want to avoid that. Would this work for you?


            utils.tryCatch(function() {
                // Find begin val
                var idxStart = data.indexOf('begin=\"');
                var idxEnd = data.indexOf('\"', idxStart);
                entry.begin = _seconds(data.substr(idxStart, idxEnd));

                // Find end val
                idxStart = data.indexOf('end=\"');
                idxEnd = data.indexOf('\"', idxStart);
                entry.end = _seconds(data.substr(idxStart, idxEnd));

                // Find text val
                idxStart = data.indexOf('\">');
                data = data.substr(idxStart + 2);
                entry.text = strings.trim(data).replace(/>\s+</g, '><');
            });